evap/evaluation/models.py

```diff
@@ -58,7 +58,7 @@ def __str__(self):
 
     @property
     def can_be_deleted_by_manager(self):
-        return all(evaluation.can_be_deleted_by_manager for evaluation in self.evaluations.all())
+        return self.evaluations.count() == 0 or (self.participations_are_archived and self.grade_documents_are_deleted and self.results_are_archived)
```

Comment:
now there's the slight inconsistency that can_be_deleted_by_manager returns false on an evaluation in a semester that can be deleting, and by deleting the semester a manager could in fact delete the evaluation. not sure what to do about that. or in other words, why (on a conceptual level) does Semester.can_be_deleted_by_manager not check Evaluation.can_be_deleted_by_manager of every evaluation?

Reply:
we could check that every evaluation can be deleted and that the archiving stuff is done, no problem. so just adding the removed part in addition?

Reply:
that would need further changes, since Evaluation.can_be_deleted_by_manager  currently checks can_be_edited which is false for archived evaluations (not sure atm whether we want that and i need to go to bed now :) )

Reply:
if we want to delete a complete semester and all of its evaluations we probably don't care about the current state of each individual evaluation. there are two "real" use cases i can think of:
1. a manager wants to delete a semester because creating it was a mistake, something went wrong, it was just a test, ...
2. all the data from a semester shall be deleted because of data privacy reasons, deleting regulations after a certain time, or similar.

deleting a semester is made "hard" in the ui because all archiving steps must have been applied (which already delete some data, so managers should be aware what they are doing) and then this explicit confirmation thing must be dealt with - so mistakenly deleting data is more or less not a problem.
that means the managers really want to have the data removed from the platform and in this situation they don't really care whether some evaluations are not finished or whether some results are not published and so on. so i would argue it's okay to just delete everything without further detailed checks.